### ISSUE TITLE
Catch illegal TS module syntax in semantic

### ISSUE THREAD

**Issue by overlookmotel** [oxc-project/oxc#3498]
Babel parser test case `typescript/module-namespace/head/input.ts`:

```ts
namespace N {}
namespace M.N.O {}
module M {}
module "m" {}
```

`module "m" {}` is illegal syntax - string IDs are only valid for ambient modules e.g. `declare module "react-dom" {}`.

Semantic is currently giving this a pass. Presumably it should reject it?

**Comment by Boshen** [comment]
Syntax check TypeScript grammar is a best effort right now, but we can add this check.

**Comment by Boshen** [comment]
This test case throws 3 errors in TypeScript, babel only throws 1, I give up. https://github.com/babel/babel/blob/main/packages/babel-plugin-transform-typescript/test/fixtures/namespace/ambient-module-nested-exported/input.ts

**Comment by Boshen** [comment]
Babel doesn't throw this error https://github.com/babel/babel/blob/main/packages/babel-parser/test/fixtures/typescript/module-namespace/head/output.json

**Comment by Boshen** [comment]
I give up, too many different conditions to check.

**Comment by Boshen** [comment]
For the record: in checker.ts, need to port `checkModuleDeclaration` otherwise false positives will arise.
